### DISCUSSION TITLE
How to exclude a method from assembly advice

### DISCUSSION THREAD

**Discussion by jorgenbosman** [ArxOne/MrAdvice#203] [Q&A] [answered]
I have my own attribute class set at assembly level (via Directory.Build.Props file), so it it set on all projects automatically without coding. I have added another constructor with an extra argument to my attribute and added that to a method. However it seems that my new constructor is called, but also the default constructor. How can I tell MrAdvice to skip the default constructor and use only the one with the extra argument?

[AttributeUsage(AttributeTargets.All, AllowMultiple = false, Inherited = true)]
public class MyAttribute : Attribute, IMethodAsyncAdvice
{
    public MyAttribute() : base() { exceptions = null; }
    public MyAttribute(Type[] exceptions) : base()
    {
        this.exceptions = exceptions;
    }
}

My method:

[MyAttribute(exceptions: [typeof(SomeException)])]
public void ThrowSomeException(Exception e)
{
}

**Comment by picrap** [comment, marked as answer]
Hi,

there is an `[ExcludePointcut]` attribute that you can apply at any method or type to exclude it from being advised.

**Comment by jorgenbosman** [comment]
It will only let me apply this on class or struct level, not on a method?

Attribute 'ExcludePointcut' is not valid on this declaration type. It is only valid on 'class, struct' declarations.

**Comment by picrap** [comment]
Oops ! My mistake, that attribute does not work this way.
Currently there is no way to explicitly exclude one method from being advised.
